src/app/profile-edit/profile-edit.page.ts

```diff
@@ -0,0 +1,139 @@
+import urlConfig from 'src/app/config/url.config.json';
+import { FETCH_Profile_FORM } from '../core/constants/formConstant';
+import { MainFormComponent } from 'elevate-dynamic-form';
+@Component({
+  selector: 'app-profile-edit',
+  templateUrl: './profile-edit.page.html',
+  styleUrls: ['./profile-edit.page.scss'],
+})
+export class ProfileEditPage implements OnInit {
+  @ViewChild('formLib') formLib: MainFormComponent | undefined;
+  formJson: any = [];
+  urlProfilePath = urlConfig['profileListing']
+
+  constructor(
+    private apiBaseService: ApiBaseService,
+    private loader: LoaderService,
+    private toastService: ToastService
+  ) {}
+
+  ngOnInit() {
+    this.getFormJson();
+  }
+
+  async getFormJson() {
+    await this.loader.showLoading("Please wait while loading...");
+    this.apiBaseService.post(urlConfig['formListing'].listingUrl, FETCH_Profile_FORM)
+      .pipe(
+        finalize(() => this.loader.dismissLoading()),
+        catchError(err => {
+          this.toastService.presentToast(err?.error?.message);
+          return of(null);
+        })
+      )
+      .subscribe((res:any) => {
+        if (res?.status === 200) {
+          this.formJson = res?.result?.data;
+          this.formJson.forEach((control: any) => {
+            if (control.dynamicEntity) {
+              this.getOptionsData(control.name);
+            }
+          });
+        }
+      });
+  }
+
+  getOptionsData(entityType: string, entityId?: string) {
+    const control = this.formJson.find((control: any) => control.name === entityType);
+    if (!control) return;
+    const urlPath = control.dynamicEntity
+      ? this.urlProfilePath.entityUrl+`${entityType}`
+      : this.urlProfilePath.subEntityUrl+`/${entityId}?type=${entityType}&search=&page=1&limit=100`;
+
+    this.apiBaseService.get(urlConfig['entityListing'].listingUrl + urlPath)
+      .pipe(
+        catchError(err => {
+          this.toastService.presentToast(err?.error?.message);
+          return of(null);
+        })
+      )
+      .subscribe((res:any) => {
+        if (res?.status === 200) {
+          const result = control.dynamicEntity ? res?.result : res?.result?.data;
+          if(result){
+            const options = result.map((entity: any) => ({
+              label: entity.name,
+              value: entity._id,
+            }));
+            this.updateFormOptions(entityType, options);
+          }
+        } else {
+          this.toastService.presentToast(res.message);
+        }
+      });
+  }
+
+  updateFormOptions(entityType: string, options: any) {
+    const control = this.formJson.find((control: any) => control.name === entityType);
+    if (control) {
+      control.options = options;
+    }
+  }
+
+  onOptionChange(event: any) {
+    const { event: selectedEvent, control } = event;
+    const selectedValue = selectedEvent?.value;
+    this.updateFormValue(control.name, selectedValue?.value);
+    this.formJson.forEach((formControl: any) => {
```

Comment:
Instead of forEach can we use map since it won't have any side effects on the formJson